### PR TITLE
Adding GetPathSafe for checking project reference paths that may throw

### PR DESCRIPTION
**Pull request by emgarten**
When finding the project closure NuGet walks through all child references to find project.json files in child projects. For native projects where the reference does not exist on disk DTE throws a generic exception.

https://github.com/NuGet/Home/issues/1110

//cc @yishaigalatzer @pranavkm @deepakaravindr

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
I don't agree with this fix. It just hides problems, we should ask the vs folks to fix it.

**Comment by emgarten** [comment]
@yishaigalatzer we should do both

File: src/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs [hunk 1]
```diff
@@ -183,5 +185,21 @@ private static IEnumerable<Reference> GetProjectReferences(EnvDTEProject project
                 }
             }
         }
+
+        private static string GetPathSafe(Reference childReference)
+        {
+            string path = null;
+
+            try
```
Comment (pranavkm) [inline src/PackageManagement.VisualStudio/ProjectSystems/BuildIntegratedProjectSystem.cs]:
```
try
{
   return childReference.Path;
}
catch
{
   return null;
}
```

**Comment by pranavkm** [comment]
I think there's a bunch of places NuGet tends to try-catch around properties VS spontaneously throws. Fixing it in VS is nice, but you're still stuck with people who don't update.

:shipit:

**Comment by deepakaravindr** [comment]
:shipit:

**Pull request merged**